pkg/cmd/server/api/latest/helpers.go

```diff
@@ -83,5 +85,25 @@ func ReadYAMLFile(filename string, obj runtime.Object) error {
 	if err != nil {
 		return err
 	}
-	return Codec.DecodeInto(data, obj)
+	err = Codec.DecodeInto(data, obj)
+	return captureSurroundingJSONForError(fmt.Sprintf("could not load config file %q due to a error: ", filename), data, err)
+}
+
+func captureSurroundingJSONForError(prefix string, data []byte, err error) error {
+	if syntaxErr, ok := err.(*json.SyntaxError); err != nil && ok {
```

Comment:
If this works, I like it.  Should we wrap json.SyntaxError in another json.SyntaxError with a a better message to preserve the error type for `CheckErr`?